### PR TITLE
Pal 649 policy service redis

### PR DESCRIPTION
**Pull request by pd104923**

### CODE REVIEW

File: policy-service-integration/src/test/java/uk/gov/gchq/palisade/integrationtests/policy/RedisPolicyCachingTest.java [hunk 1]
```diff
@@ -91,57 +84,56 @@ public void addedPolicyIsRetrievable() {
             Optional<Policy> policy = cacheProxy.getPolicy(resource);
 
             // Then
-            assertTrue(policy.isPresent());
+            assertThat(policy).isNotNull();
```
Comment (dbz120896) [inline policy-service-integration/src/test/java/uk/gov/gchq/palisade/integrationtests/policy/RedisPolicyCachingTest.java:87]: Should this be `.isPresent()`?

**Review by dbz120896 — CHANGES_REQUESTED** [review]

File: policy-service-integration/src/test/java/uk/gov/gchq/palisade/integrationtests/policy/RedisPolicyCachingTest.java (hunk 1 above)
Comment (developer01189998819991197253) [reply policy-service-integration/src/test/java/uk/gov/gchq/palisade/integrationtests/policy/RedisPolicyCachingTest.java:87]: .isPresent is the Junit way of doing not null checks using just Junit, and assertThat(x).isNotNull is the assertJ way which we agreedupon in the high level test strategy

Comment (dbz120896) [reply policy-service-integration/src/test/java/uk/gov/gchq/palisade/integrationtests/policy/RedisPolicyCachingTest.java:87]: But, doesn't `cacheProxy.getPolicy(resource);` always return a non-null optional? So, policy will always be non-null.

**Review by dbz120896 — DISMISSED** [review]

File: policy-service-integration/src/test/java/uk/gov/gchq/palisade/integrationtests/policy/RedisPolicyCachingTest.java [hunk 2]
```diff
@@ -60,88 +53,87 @@
...
-            e.printStackTrace();
-        }
-        Optional<Policy> cachedPolicy = cacheProxy.getPolicy(makeResource.apply(0));
+        // When
+        Optional<Policy> returnedPolicy = cacheProxy.getPolicy(systemResource);
 
-        // Then - it has been evicted
-        assertTrue(cachedPolicy.isEmpty());
+        // Then the returned policy should have the updated resource rules
+        assertThat(returnedPolicy.get()).isNotNull();
```
Comment (dev930018) [inline policy-service-integration/src/test/java/uk/gov/gchq/palisade/integrationtests/policy/RedisPolicyCachingTest.java:121]: There should be no need to test Optional.get() is not null, but rather should test Optional.isPresent()

File: policy-service-integration/src/test/java/uk/gov/gchq/palisade/integrationtests/policy/CaffeinePolicyCachingTest.java [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+    public void testAddedPolicyIsRetrievable() {
+        // Given - resources have been added as above
+        // Given there is no underlying policy storage (gets must be wholly cache-based)
+
+        for (Resource resource : FILE_RESOURCES) {
+            // When
+            Policy policy = policyService.getPolicy(resource).get();
+
+            // Then
+            assertThat(policy).isNotNull();
```
Comment (dev930018) [inline policy-service-integration/src/test/java/uk/gov/gchq/palisade/integrationtests/policy/CaffeinePolicyCachingTest.java:90]: Ditto some optional isPresent check over isNotNull

**Review by dev930018 — CHANGES_REQUESTED** [review]

File: policy-service-integration/src/test/java/uk/gov/gchq/palisade/integrationtests/policy/config/RedisTestConfiguration.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.boot.autoconfigure.data.redis.RedisProperties;
+import org.springframework.context.annotation.Configuration;
+import redis.embedded.RedisServer;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+
+@Configuration
+@EnableAutoConfiguration
```
[the remark this answers is not in the document]
Comment (developer01189998819991197253) [reply policy-service-integration/src/test/java/uk/gov/gchq/palisade/integrationtests/policy/config/RedisTestConfiguration.java:28]: I believe so, it was implemeted in the user service redis testing also

**Review by dev930018 — APPROVED** [review]

**Review by ac74475 — APPROVED** [review]

**Review by dev930018 — APPROVED** [review]
Aside from possibly moving to Palisade-services/policy-service/src/component-test

**Comment by developer01189998819991197253** [comment]
> 
> 
> Aside from possibly moving to Palisade-services/policy-service/src/component-test

For existing tickets we can leave them as is, in the future we will transition

**Review by developer01189998819991197253 — APPROVED** [review]

**Review by ac74475 — APPROVED** [review]

**Pull request merged**